src/battle.ts

```diff
@@ -3292,6 +3292,18 @@ class Battle {
 			}
 			break;
 		}
+		case 'updatepokemon': {
+			const {siden} = this.parsePokemonId(args[1]);
+			const side = this.sides[siden];
+			for (let i = 0; i < side.pokemon.length; i++) {
+				const pokemon = side.pokemon[i];
+				if (pokemon.details === args[2]) {
+					side.addPokemon('', '', args[3], i);
```

Comment:
```suggestion
				if (pokemon.checkDetails(args[2])) {
					side.addPokemon('', '', args[2], i);
```

Reply:
In theory, this should be all you need to make it support `|updatepokemon|POKEMONID|NEWDETAILS`

Reply:
this worked for the dogs but not xerneas, presumably because it's getting "Neutral" added on rather than "-*". If xerneas was changed to have neutral as the base form this could probably work (but that would be a big hassle).

Reply:
Hm. There are a few ways around that.

Reply:
The simplest way is to send `-*`, and then hardcode Neutral form in Team Preview.